source/opt/ilist_node.h

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef LIBSPIRV_OPT_ILIST_NODE_H_
+#define LIBSPIRV_OPT_ILIST_NODE_H_
+
+#include <cassert>
+
+namespace spvtools {
+namespace utils {
+
+template <class NodeType>
+class IntrusiveList;
+
+// IntrusiveNodeBase is the class when creating classes than can be inserted
```

Comment:
Fixed the comment.

I can change the two spaces at the end of sentences.  Most style guides say 1 space, but that is because generally people don't use monospace fonts.  Not us programmers.

Reply:
I'm pretty old school.  I learned touch-typing on a mechanical typewriter and am in the habit of two spaces after a period.  I'm ok with others doing it either single-space or double-space.